### PR TITLE
NIFI-13635 Expose processor bulletins as a part of FlowInfo

### PR DESCRIPTION
**Pull request by pkedvessy**
# Summary

[NIFI-13635](https://issues.apache.org/jira/browse/NIFI-13635)

# Tracking

Please complete the following tracking steps prior to pull request creation.

### Issue Tracking

- [x] [Apache NiFi Jira](https://issues.apache.org/jira/browse/NIFI) issue created

### Pull Request Tracking

- [x] Pull Request title starts with Apache NiFi Jira issue number, such as `NIFI-00000`
- [x] Pull Request commit message starts with Apache NiFi Jira issue number, as such `NIFI-00000`

### Pull Request Formatting

- [x] Pull Request based on current revision of the `main` branch
- [x] Pull Request refers to a feature branch with one commit containing changes

# Verification

Please indicate the verification steps performed prior to pull request creation.

### Build

- [x] Build completed using `mvn clean install -P contrib-check`
  - [x] JDK 21

### Licensing

- [ ] New dependencies are compatible with the [Apache License 2.0](https://apache.org/licenses/LICENSE-2.0) according to the [License Policy](https://www.apache.org/legal/resolved.html)
- [ ] New dependencies are documented in applicable `LICENSE` and `NOTICE` files

### Documentation

- [ ] Documentation formatting appears as expected in rendered files

### CODE REVIEW

File: c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/ProcessorBulletin.java [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+ * limitations under the License.
+ */
+
+package org.apache.nifi.c2.protocol.api;
+
+import io.swagger.v3.oas.annotations.media.Schema;
+import java.io.Serializable;
+import java.util.Date;
+
+public class ProcessorBulletin  implements Serializable {
```
Comment (ferencerdei) [inline c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/ProcessorBulletin.java:24]: There is an extra space here after the class name

File: c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/ProcessorBulletin.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+    private String category;
+    private String message;
+    private String groupId;
+    private String groupName;
+    private String groupPath;
+    private String sourceId;
+    private String sourceName;
+    private String flowFileUuid;
+
+    @Schema(description = "When this bulletin was generated.")
```
Comment (ferencerdei) [inline c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/ProcessorBulletin.java:41]: It seems the convention in the C2 API is to not have dots at the end of the descriptions. Could you remove these?

**Review by ferencerdei — CHANGES_REQUESTED** [review]
Thanks for making this possible in the C2 protocol, it will help users to debug flows without needing to SSH to the specific agent to see the logs.

I have only 2 minor comments, please have a look.

File: c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/ProcessorBulletin.java (hunk 2 above)
Comment (pkedvessy) [reply c2/c2-protocol/c2-protocol-api/src/main/java/org/apache/nifi/c2/protocol/api/ProcessorBulletin.java]: Done

**Pull request closed without merging**